test/unit/SES/Transport/SESTransportTest.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * @author    Craig Gosman <craig@ingenerator.com>
+ * @licence   proprietary
+ */
+
+namespace test\unit\Ingenerator\SwiftMailer\SES\Transport;
+
+use Ingenerator\PHPUtils\Monitoring\ArrayMetricsAgent;
+use Ingenerator\PHPUtils\Monitoring\OperationTimer;
+use Ingenerator\SwiftMailer\SES\Transport\SESTransport;
+use InvalidArgumentException;
+use PHPUnit\Framework\Assert;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Log\LogLevel;
+use Psr\Log\Test\TestLogger;
+use Swift_Events_EventDispatcher;
+use Swift_Events_SendEvent;
+use Swift_Events_SendListener;
+use Swift_Events_SimpleEventDispatcher;
+use Swift_Message;
+use Swift_Mime_SimpleMessage;
+use test\mock\Ingenerator\SwiftMailer\SES\Transport\SpyingSESClient;
+
+class DummySwiftPlugin implements Swift_Events_SendListener
+{
+    protected array $got_events;
+
+    public function assertGotEvents(array $expect)
+    {
+        Assert::assertSame($expect, $this->got_events);
+    }
+
+    public function beforeSendPerformed(Swift_Events_SendEvent $evt)
+    {
+        $this->got_events[] = [__FUNCTION__, $evt];
+    }
+
+    public function sendPerformed(Swift_Events_SendEvent $evt)
+    {
+        $this->got_events[] = [__FUNCTION__, $evt];
+    }
+
+}
+
+class DummySwift_Message extends Swift_Message
+{
+    private string $dummy_mime_string;
+
+    public static function returnDummyMimeString(string $string): DummySwift_Message
```

Comment:
Very minor thing - `returnXXX` at first glance to me looks like the method that does the returning. `withDummyMimeString` closer to our normal naming for mock/dummy constructors.